### DISCUSSION TITLE
How to compose source files from the memory like `@rollup/plugin-virtual`?

### DISCUSSION THREAD

**Discussion by samchon** [swc-project/swc#8756] [Q&A]
https://www.npmjs.com/package/@rollup/plugin-virtual

I hope to utilize the `@swc/wasm-web` with multiple source files that is composed in the memory.

How can I do that? Finding types defined in the `@swc/wasm-web`, but don't know which option is for it.

```typescript
const output = await swc.transform(
  `import { plus } from "./plus";\n\nconsole.log(plus(3, 4));`, 
  {
    files: {
      "./plus.ts": "export const plus = (x: number, y: number) => x + y",
    },
  },
);
```

**Comment by kwonoj** [comment]
I don't believe swc/wasm is equivalent to the plugin you want to achieve. After all, current swc's transform is 1:1 input - ouput fn, so if you need to provide some features on top of it you may need to construct your own.

**Reply by samchon** [reply]
Got it. Thanks for fast replying.
